frontend/src/metabase/query_builder/components/expressions/ExpressionWidget.tsx

```diff
@@ -0,0 +1,124 @@
+  onRemoveExpression?: (name: string) => void;
+  onClose?: () => void;
+}
+
+const ExpressionWidget = (props: ExpressionWidgetProps): JSX.Element => {
+  const {
+    query,
+    name: initialName,
+    expression: initialExpression,
+    reportTimezone,
+    onChangeExpression,
+    onRemoveExpression,
+    onClose,
+  } = props;
+
+  const [name, setName] = useState(initialName || "");
+  const [expression, setExpression] = useState<Expression | null>(
+    initialExpression || null,
+  );
+  const [error, setError] = useState<string | null>(null);
+
+  const helpTextTargetRef = useRef(null);
+
+  const isValid = !!name && !error && isExpression(expression);
+
+  const handleCommit = () => {
+    if (isValid && isDef(expression)) {
+      onChangeExpression(name, expression);
+      onClose && onClose();
+    }
+  };
+
+  return (
+    <Wrapper>
+      <div className="p2">
+        <div className="h5 text-uppercase text-light text-bold">{t`Expression`}</div>
+        <div ref={helpTextTargetRef}>
+          <ExpressionEditorTextfield
+            helpTextTarget={helpTextTargetRef.current}
+            expression={expression}
+            name={name}
+            query={query}
+            reportTimezone={reportTimezone}
+            onChange={(parsedExpression: Expression) => {
+              setExpression(parsedExpression);
+              setError(null);
+            }}
+            onError={(errorMessage: string) => setError(errorMessage)}
+          />
+        </div>
+
+        <div className="mt3 h5 text-uppercase text-light text-bold">{t`Name`}</div>
+        <div>
+          <input
+            className="my1 input block full"
+            type="text"
+            value={name}
+            placeholder={t`Something nice and descriptive`}
+            onChange={event => setName(event.target.value)}
+            onKeyPress={e => {
+              if (e.key === "Enter") {
+                handleCommit();
+              }
+            }}
+          />
+        </div>
+      </div>
+
+      <div className="mt2 p2 border-top flex flex-row align-center justify-between">
+        <div className="ml-auto">
+          <button
+            className="Button"
+            onClick={() => onClose && onClose()}
+          >{t`Cancel`}</button>
+          <button
+            className={cx("Button ml2", {
+              "Button--primary": isValid,
+            })}
+            onClick={handleCommit}
+            disabled={!isValid}
+          >
+            {initialName ? t`Update` : t`Done`}
+          </button>
+        </div>
+        <div>
+          {initialName && onRemoveExpression ? (
+            <a
+              className="pr2 ml2 text-error link"
+              onClick={() => {
+                onRemoveExpression(initialName);
+                onClose && onClose();
+              }}
+            >{t`Remove`}</a>
+          ) : null}
+        </div>
+      </div>
+    </Wrapper>
+  );
+};
+
+const Wrapper = styled.div`
```

Comment:
We create `<filename>.styled.tsx` files for styled components. Here it would be `ExpressionWidget.styled.tsx`

Reply:
It certainly makes sense to me, but that's not the point. The thing is that we have an existing pattern to extract styled components to a separate file and we follow it (hopefully) everywhere. If this pattern has flaws, we can discuss it during the FE meeting, but until there is consensus about this, we should not introduce new patterns. Does it make sense?